scripts/qa_crowbarsetup.sh

```diff
@@ -850,31 +850,26 @@ function cluster_node_assignment()
             if crowbar machines show "$node" | grep "\"macaddress\"" | grep -qi $mac ; then
                 nodesavailable=`remove_node_from_list "$node" "$nodesavailable"`
                 clusternodesdrbd="$clusternodesdrbd $node"
+                echo "Claiming disk for DRBD on node: $node"
 
                 # assign drbd volume via knife
-                local nfile
-                nfile=knife.node.${node}.json
-                knife node show ${node} -F json > $nfile
-                rubyjsonparse "
-                            j['normal']['crowbar_wall']['claimed_disks'].each do |k,v|
+                knife exec -E "
+                    nodes.find(:name => '${node}').each do |n|
+                        if n['crowbar_wall']['claimed_disks']
+                            n['crowbar_wall']['claimed_disks'].each do |k,v|
                                 next if v.is_a? Hash and v['owner'] !~ /LVM_DRBD/;
-                                j['normal']['crowbar_wall']['claimed_disks'].delete(k);
-                            end ;
-                            j['normal']['crowbar_wall']['claimed_disks']['/dev/disk/by-id/$(get_disk_id_by_serial_and_libvirt_type "$libvirt_type" "$serial")']={'owner' => 'LVM_DRBD'};
-                            puts JSON.pretty_generate(j)" < $nfile > ${nfile}.tmp
-                mv ${nfile}.tmp ${nfile}
-                knife node from file ${nfile}
+                                n['crowbar_wall']['claimed_disks'].delete(k);
+                            end
+                        else
+                            n['crowbar_wall']['claimed_disks'] = {}
+                        end
+                        n['crowbar_wall']['claimed_disks']['/dev/disk/by-id/$(get_disk_id_by_serial_and_libvirt_type "$libvirt_type" "$serial")']={'owner' => 'LVM_DRBD'};
+                        n.save
+                    end
+                "
             fi
         done
     done
-    for dnode in $clusternodesdrbd ; do
```

Comment:
Yes. Right it's not needed anymore, because "knife exec" doesn't drop the automatic attributes.